src/main/java/com/zaxxer/hikari/metrics/prometheus/PrometheusMetricsTrackerFactory.java

```diff
@@ -19,29 +19,89 @@
 import com.zaxxer.hikari.metrics.IMetricsTracker;
 import com.zaxxer.hikari.metrics.MetricsTrackerFactory;
 import com.zaxxer.hikari.metrics.PoolStats;
+import io.prometheus.client.CollectorRegistry;
 
 /**
+ * <p>
+ * Use default CollectorRegistry and default HikariCPCollector:
  * <pre>{@code
  * HikariConfig config = new HikariConfig();
  * config.setMetricsTrackerFactory(new PrometheusMetricsTrackerFactory());
  * }</pre>
+ * </p>
+ * <p>
+ * Use provided CollectorRegistry and default HikariCPCollector:
+ * <pre>{@code
+ * HikariConfig config = new HikariConfig();
+ * config.setMetricsTrackerFactory(new PrometheusMetricsTrackerFactory(registry));
+ * }</pre>
+ * </p>
+ * <p>
+ * Use provided CollectorRegistry and provided HikariCPCollector:
+ * <pre>{@code
+ * HikariConfig config = new HikariConfig();
+ * config.setMetricsTrackerFactory(new PrometheusMetricsTrackerFactory(registry, collector));
+ * }</pre>
+ * </p>
  */
 public class PrometheusMetricsTrackerFactory implements MetricsTrackerFactory {
 
-   private static HikariCPCollector collector;
+   private HikariCPCollector collector;
+   private CollectorRegistry registry;
+
+   private static HikariCPCollector globalCollector;
+
+   /**
+    * If nothing is provided, use {@link CollectorRegistry.defaultRegistry} and {@link globalCollector} as default
+    */
+   public PrometheusMetricsTrackerFactory() {
+      this(CollectorRegistry.defaultRegistry);
+   }
+
+   /**
+    * If a CollectorRegistry is provided, use provided CollectorRegistry and {@link globalCollector} as default
+    */
+   public PrometheusMetricsTrackerFactory(CollectorRegistry registry) {
+      if (globalCollector == null) {
+         synchronized (PrometheusMetricsTrackerFactory.class) {
+            if (globalCollector == null) {
+               globalCollector = new HikariCPCollector();
+            }
+         }
+      }
```

Comment:
[Broken Double-checked locking](https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java) pattern.

Reply:
I resolved this problem by adding volatile keyword for the filed.